### DISCUSSION TITLE
Uvicorn can't work with reload option and debug mode in PyCharm

### DISCUSSION THREAD

**Discussion by yakimka** [Kludex/uvicorn#2118] [Q&A] [answered]
I am currently unsure whether the issue I encountered is attributed to Uvicorn or PyCharm. 
I found a problem when I use PyCharm debugging with Uvicorn reload mode, and Python in development mode.

When I run debug in PyCharm with the command `uvicorn main:app --reload` and with the `PYTHONDEVMODE=1` environment variable ([link to docs](https://docs.python.org/3/library/devmode.html)), I get this error:

```
unknown option --port
usage: /usr/local/bin/python3 [option] ... [-c cmd | -m mod | file | -] [arg] ...
Try `python -h' for more information.
```

I made a simple setup to show this problem. You can see it here: https://github.com/yakimka/pycharm_debug_with_devmode_bug

Can anyone help me with this?

**Comment by Kludex** [comment, marked as answer]
Already raised by #2000.

**Reply by yakimka** [reply]
Oh, I saw this issue but wasn't sure if it was relevant. Thank you.
